tests/simulator/__init__.py

```diff
@@ -84,9 +112,9 @@ def makeBasicModel(extra_pops={}):
             "bz": lambda x: np.cos(2 * pi_over_xmax * x),
         },
         2: {
-            "bx": lambda x, y: np.cos(2 * pi_over_xmax * x),
-            "by": lambda x, y: np.sin(1 * pi_over_xmax * x),
-            "bz": lambda x, y: np.cos(2 * pi_over_xmax * x),
+            "bx": lambda x, y: fn_2d_periodic(sim, x, y),
```

Comment:
I realise that for this test to be better it's probably more appropriate to change a bit the values of each component, that avoids bugs like dumping By at the place of Bx to be unnoticed.
I would simply add a constant parameter to `fn_2d_periodic` and add that constant to the returned value so each B and V component have a distinct value